<a href="https://colab.research.google.com/github/ZachGermain/DSPS_zgermain/blob/master/HW3_zgermain/Z_testlab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
%pylab inline

Populating the interactive namespace from numpy and matplotlib


I extracted precipitations records for Willmington from 2000/01/01 through today. 
The goal is to extract from these data the daily precipitation starting on 2018/01/01 and assess if these data are different from a population with mean $\mu = 0.14$ standard deviation $\sigma= 0.38$. 

Set a p-value at $2\sigma$:  $p=0.05$

In [0]:
rain = pd.read_csv("https://raw.githubusercontent.com/fedhere/DSPS/master/labs/1865331.csv")


The $Z$ statistics is : $Z = \frac{\bar{X} - \mu }{s}$ and $s = \sigma / \sqrt(N)$ (you should know what all those symbols mean)

1. Extract from the dataframe above the columns "DATE" and "PRCP" (precipitation)

2. Convert the dataframe DATE from ```string``` to ```datetime``` type
 using pd.datetime() 
    
3. Select the data rows with date later than 2018/01/01. 
    If your data type conversion worked well you should be able to set the condition as ```rain[<newcolumnname>]> pd.datetime("2018/01/01")```
    use the technique called "broadcasting" to select the rows: 
    
    ```rain[ rain[<newcolumnname>]> pd.datetime("2018/01/01")]``` which you can save as a new dataframe
    
4. Calculate the mean of the sample you created $\bar{X}$ and plug it in  the numerator of the $Z$ statistics: $\bar{X}- \mu$

5. Calculate the size of the sample and plug it in the formula. Now you have $Z$

6. Knowing the distribution of $Z\sim N(\mu=0, \sigma=1)$ and with a confidence $p=0.05$, did you find that the time period 2018/01/01---today has a statistically significant different amount of rain?  
    

In [0]:
#This section describes the precipitation data
datePRCPCols = rain[ ["DATE", "PRCP"] ]
datePRCPCols["date"] = pd.to_datetime(datePRCPCols["DATE"])
rows2018Plus = datePRCPCols[ datePRCPCols["date"]> pd.to_datetime("2018/01/01") ]
rows2018Plus["PRCP"].describe()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


count    615.000000
mean       0.162504
std        0.365007
min        0.000000
25%        0.000000
50%        0.000000
75%        0.120000
max        2.450000
Name: PRCP, dtype: float64

In [0]:
#This calculates the Z-score
numZ = 0.16 - 0.14
denZ = 0.38/np.sqrt(615)
numZ/denZ

1.3052207123828672

$p < 0.05$ implies that $Z > 2$. We found $Z = 1.3$ which means that we have not found that from 2018 until today, a significant difference in precipitation compared to other times.